src/widgets/status_icon.rs

```diff
@@ -0,0 +1,242 @@
+    }
+
+    pub fn get_tooltip_text(&self) -> Option<String> {
+        unsafe {
+            from_glib_full(ffi::gtk_status_icon_get_tooltip_text(self.to_glib_none().0))
+        }
+    }
+
+    pub fn get_visible(&self) -> bool {
+        unsafe {
+            from_glib(ffi::gtk_status_icon_get_visible(self.to_glib_none().0))
+        }
+    }
+
+    pub fn get_x11_window_id(&self) -> u32 {
+        unsafe {
+            ffi::gtk_status_icon_get_x11_window_id(self.to_glib_none().0)
+        }
+    }
+
+    pub fn is_embedded(&self) -> bool {
+        unsafe {
+            from_glib(ffi::gtk_status_icon_is_embedded(self.to_glib_none().0))
+        }
+    }
+
+    pub fn set_from_file(&self, filename: &str) {
+        unsafe {
+            ffi::gtk_status_icon_set_from_file(self.to_glib_none().0, filename.to_glib_none().0);
+        }
+    }
+
+    //pub fn set_from_gicon(&self, icon: Unknown rust type: "Icon") {
+    //    unsafe { TODO: call ffi:gtk_status_icon_set_from_gicon() }
+    //}
+
+    pub fn set_from_icon_name(&self, icon_name: &str) {
+        unsafe {
+            ffi::gtk_status_icon_set_from_icon_name(self.to_glib_none().0, icon_name.to_glib_none().0);
+        }
+    }
+
+    pub fn set_from_pixbuf(&self, pixbuf: Option<&gdk::pixbuf::Pixbuf>) {
+        unsafe {
+            ffi::gtk_status_icon_set_from_pixbuf(self.to_glib_none().0, pixbuf.to_glib_none().0);
+        }
+    }
+
+    pub fn set_from_stock(&self, stock_id: &str) {
+        unsafe {
+            ffi::gtk_status_icon_set_from_stock(self.to_glib_none().0, stock_id.to_glib_none().0);
+        }
+    }
+
+    pub fn set_has_tooltip(&self, has_tooltip: bool) {
+        unsafe {
+            ffi::gtk_status_icon_set_has_tooltip(self.to_glib_none().0, has_tooltip.to_glib());
+        }
+    }
+
+    pub fn set_name(&self, name: &str) {
+        unsafe {
+            ffi::gtk_status_icon_set_name(self.to_glib_none().0, name.to_glib_none().0);
+        }
+    }
+
+    pub fn set_screen(&self, screen: &gdk::Screen) {
+        unsafe {
+            ffi::gtk_status_icon_set_screen(self.to_glib_none().0, screen.to_glib_none().0);
+        }
+    }
+
+    pub fn set_title(&self, title: &str) {
+        unsafe {
+            ffi::gtk_status_icon_set_title(self.to_glib_none().0, title.to_glib_none().0);
+        }
+    }
+
+    pub fn set_tooltip_markup(&self, markup: Option<&str>) {
+        unsafe {
+            ffi::gtk_status_icon_set_tooltip_markup(self.to_glib_none().0, markup.to_glib_none().0);
+        }
+    }
+
+    pub fn set_tooltip_text(&self, text: &str) {
+        unsafe {
+            ffi::gtk_status_icon_set_tooltip_text(self.to_glib_none().0, text.to_glib_none().0);
+        }
+    }
+
+    pub fn set_visible(&self, visible: bool) {
+        unsafe {
+            ffi::gtk_status_icon_set_visible(self.to_glib_none().0, visible.to_glib());
+        }
+    }
+
+    pub fn position_menu(menu: &::Menu, x: &mut i32, y: &mut i32, user_data: &StatusIcon) -> bool {
```

Comment:
I'd comment this one out because it's pointless with our limited version of `popup`.